sw/device/lib/dif/dif_uart.c

```diff
@@ -472,3 +472,58 @@ dif_uart_result_t dif_uart_tx_bytes_available(const dif_uart_t *uart,
 
   return kDifUartOk;
 }
+
+dif_uart_result_t dif_uart_fifo_reset(const dif_uart_t *uart,
+                                      dif_uart_fifo_reset_t reset) {
+  if (uart == NULL) {
+    return kDifUartBadArg;
+  }
+
+  uint32_t reg = mmio_region_read32(uart->base_addr, UART_FIFO_CTRL_REG_OFFSET);
+  bitfield_field32_t field = {
+      .mask = 0x1, .value = 1,
+  };
+
+  if (reset == kDifUartFifoResetRx || reset == kDifUartFifoResetAll) {
+    field.index = UART_FIFO_CTRL_RXRST;
+    reg = bitfield_set_field32(reg, field);
+  }
+
+  if (reset == kDifUartFifoResetTx || reset == kDifUartFifoResetAll) {
+    field.index = UART_FIFO_CTRL_TXRST;
+    reg = bitfield_set_field32(reg, field);
+  }
```

Comment:
I feel like this could be writen more concisely as a switch-with-fallthrough. Up to you though. I might also consider not pre-declaring field, since it means that the three bitfield parameters are visually split up. (Not that it's a huge deal, once Sam's bitfield.h refactor is done...).

Reply:
I actually have warmed up to inlining fields, so Ill fix up that, and have a look at switch suggestion.

Reply:
Didn't find a good way of doing the `switch`, so went with the original, but inlined the `field`.

Reply:
Maybe it was easier in my head. oh well. I might consider uniting the ifs into an if-else-if just to make it clear that they're part of the same unit though...